### ISSUE TITLE
AWS Provider v3 support

### ISSUE THREAD

**Issue by bigwheel** [cloudposse/terraform-aws-tfstate-backend#62]
## Describe the Feature

AWS Provider v3 support

## Expected Behavior

We can use aws provider v3

## Use Case

Provider update is needed by security our comaptibility with other modules.

## Describe Ideal Solution

A clear and concise description of what you want to happen. If you don't know, that's okay.

## Alternatives Considered

Stay in aws 2.x

## Upgrade Blocker

AWS 2.x to 3.x has no compatibility. In `terraform-aws-tfstate-backend`, following error occurs.

```sh
$ terraform plan -var-file="fixtures.us-west-1.tfvars"
Warning: Value for undeclared variable

The root module does not declare a variable named "s3_bucket_name" but a value
was found in file "fixtures.us-west-1.tfvars". To use this value, add a
"variable" block to the configuration.

Using a variables file to set an undeclared variable is deprecated and will
become an error in a future release. If you wish to provide certain "global"
settings to all configurations in your organization, use TF_VAR_...
environment variables to set these instead.


Error: Computed attribute cannot be set

  on ../../main.tf line 128, in resource "aws_s3_bucket" "default":
 128:   region        = var.region
```

This is written in https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/version-3-upgrade#region-attribute-is-now-read-only .

Therefore, I guess we cannot support both aws v3 and aws v2.

**Comment by bigwheel** [comment]
Finally, I think we have a version compatibility table like

| terraform-aws-tfstate-backend version | AW Provider version |
|------------|----------|
| < 0.24.0 | ~> 2.0 |
| >= 0.24.0 | ~> 3.0 |

**Comment by bigwheel** [comment]
similar: https://github.com/cloudposse/terraform-aws-s3-bucket/pull/51#issuecomment-686293744

**Comment by osterman** [comment]
This should now be addressed by #63
